docs/integration/os/yocto/lvgl_recipe.rst

```diff
@@ -0,0 +1,913 @@
+    Enter target directory for SDK (default: /opt/poky/5.0.4): /opt/poky/sdk-with-lvgl
+    You are about to install the SDK to "/opt/poky/sdk-with-lvgl". Proceed [Y/n]? y
+
+If you want to ensure the SDK was generated with lvgl being installed, go to 
+the path you extracted the SDK and find all lvgl files:
+
+.. code-block:: bash
+
+    cd /opt/poky/5.0.4/sysroots/cortexa53-poky-linux
+    find . -name "*lvgl*"
+
+The ``.so`` files you will find will depend on the LVGL configuration you used.
+
+Now to use the SDK environment and cross-compile an application:
+
+.. code-block:: bash
+
+    source /opt/poke/5.0.4/environment-setup-cortexa53-poky-linux
+
+.. note::
+
+   The tools available in LVGL library will depend on the build configuration 
+   the recipe was done with.
+
+
+Custom recipe 
+*************
+
+Until this section, everything was already done for you. We used existing 
+recipes. The objective here is to create a recipe from scratch and to add the 
+generated binary in the image.
+
+Create a layer 
+==============
+
+First, create a layer and add it to the configuration file
+
+.. code-block::
+
+   bitbake-layers create-layer ../sources/meta-mylvgl
+   bitbake-layers add-layer ../sources/meta-mylvgl
+
+In the ``sources`` folder, a new folder was created: ``meta-mylvgl`` and the 
+directory tree should look like the following
+
+.. code-block:: none
+
+   ├── conf
+   │   └── layer.conf
+   ├── COPYING.MIT
+   ├── README
+   └── recipes-example
+       └── example
+           └── example_0.1.bb
+
+
+Create a recipe 
+===============
+
+Following this structure, create  a folder containing the recipies to build 1 
+or multiple applications using lvgl
+
+.. code-block::
+   
+   cd ../sources/meta-mylvgl
+   mkdir -p recipes-lvglapp/lvgl-fbdev-benchmark/files
+   touch recipes-lvglapp/lvgl-fbdev-benchmark/lvglbenchmarkfbdev_2.4.bb
+
+We will focus on 1 application that will clone an lvgl git repository and patch 
+it for our needs.
+
+The content of ``recipes-lvglapp/lvgl-fbdev-benchmark/lvglbenchmarkfbdev_2.4.
+bb``
+
+.. code-block:: none
+   
+   DESCRIPTION = "My C++ lvgl app with CMake"
+   
+   LICENSE = "MIT"
+   LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
+
+   # Update SRC_URI to point to the GitHub repository
+   SRC_URI = "gitsm://github.com/lvgl/lv_port_linux.git;branch=master;protocol=https \
+              file://0001-change-config-file-to-run-fbdev-with-2-threads.patch \
+              file://0002-adapt-CMakeLists-file-to-compile-and-link-fbdev.patch \
+             " 
+
+   S = "${WORKDIR}/git"
+
+   inherit cmake
+
+   CMAKE_PROJECT_NAME = "lvgl_app"
+   CMAKE_PROJECT_VERSION = "2.0"
+
+   do_install() {
+       install -d ${D}${bindir}
+       install -m 0755 ${S}/bin/lvglbenchmark ${D}${bindir}
+   }
+
+The sources come from ``lv_port_linux`` repository. We apply 2 patches to modify the ``CMakeLists.txt`` and ``lv_conf.h``.
+
+Patch 1
```

Comment:
Can we fix LVGL so that these patches are not required?

Reply:
Yes, but I prefer including these patches because they demonstrate how users can apply one or more patches. Since the guide is also intended to provide some tips for Yocto, I think this is a reasonable way to illustrate the process, don’t you agree?

Reply:
> demonstrate how users can apply one or more patches.

How about fix those issues and include a demo purpose patch, say change one line of comment instead? So the code can work out of box without patching?

Reply:
I agree with that. I guess I'll wait the current PR on lv_port_linux to fix Wayland and I'll update with your suggestion.